### PR TITLE
check if the given tree is binary tree or not through recursion

### PR DESCRIPTION
**Pull request by Mohim-Singla**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

This program check if the given tree is binary tree or not through recursion.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#3337

### CODE REVIEW

File: C-Plus-Plus/ds/Check_for_Binary_search_tree.cpp [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+A BST
+A non-BST
+
+Output:
+True
+False
+
+Time complexity: O(nlogn)
+Space complexity: O(n)
+*/
```
Comment (bhav09) [inline C-Plus-Plus/ds/Check_for_Binary_search_tree.cpp:119]: Add a new at EOF

File: C-Plus-Plus/ds/Check_for_Binary_search_tree.cpp [hunk 2]
```diff
@@ -0,0 +1,117 @@
+/*
+Validating BST: This program check if the given tree is BST or not.
```
Comment (bhav09) [inline C-Plus-Plus/ds/Check_for_Binary_search_tree.cpp:2]: Can you describe a BST here. The description here is very short.

**Review by bhav09 — CHANGES_REQUESTED** [review]
Reviewed.

**Comment by Mohim-Singla** [comment]
@bhav09 Made the required changes in the program.

File: C-Plus-Plus/ds/Check_for_Binary_search_tree.cpp [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    root=create(root, 11);
+    root=create(root, 25);
+    root=create(root, 8);
+    root=create(root, 6);
+    root=create(root, 14);
+    root=create(root, 2);
+    root=create(root, 21);
+    root=create(root, 31);
+    root=create(root, 7);
+    root=create(root, 32);
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/Check_for_Binary_search_tree.cpp]: Use dynamic input

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Format your code properly as well

**Comment by Mohim-Singla** [comment]
@HarshCasper  done

File: C-Plus-Plus/ds/Check_for_Binary_search_tree.cpp [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+BST: A tree with following conditions:
+1. At max 2 childern.
+2. Value stored in left child < Value of Parent node < Value of right child. (Must be valid for every node)
+
+Author: @Mohim-Singla
+*/
+
+#include <iostream>
+#include <stdlib.h>
+
```
Comment (atarax665) [inline C-Plus-Plus/ds/Check_for_Binary_search_tree.cpp:11]: Include ```#include <limit.h>``` as you have used INT_MIN and INT_MAX. Or you can also use ```#include <bits/stdc++.h>``` directly.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by Mohim-Singla** [comment]
@atarax665 resolved.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**